dev/tools/update_icons.dart

```diff
@@ -231,20 +243,50 @@ Map<String, String> stringToTokenPairMap(String codepointData) {
 
 // Do not make this method private as it is used by g3 roll.
 String regenerateIconsFile(String iconData, Map<String, String> tokenPairMap) {
+  final Iterable<_Icon> newIcons = tokenPairMap.entries.map((MapEntry<String, String> entry) => _Icon(entry));
   final StringBuffer buf = StringBuffer();
   bool generating = false;
+  
   for (final String line in LineSplitter.split(iconData)) {
     if (!generating) {
       buf.writeln(line);
     }
-    if (line.contains(_beginGeneratedMark)) {
+
+    // Generate for _PlatformAdaptiveIcons
+    if (line.contains(_beginPlatformAdaptiveGeneratedMark)) {
       generating = true;
 
-      final String iconDeclarationsString = <String>[
-        for (MapEntry<String, String> entry in tokenPairMap.entries)
-          _Icon(entry).fullDeclaration
-      ].join();
+      final List<String> platformAdaptiveDeclarations = <String>[];
+      _platformAdaptiveIdentifiers.forEach((String flutterId, List<String> ids) {
+        // Automatically finds and generates styled icon declarations.
+        for (final IconStyle iconStyle in IconStyle.values) {
+          final String style = iconStyle.idSuffix();
+          try {
+            final _Icon agnosticIcon = newIcons.firstWhere((_Icon icon) => icon.id == '${ids[0]}$style');
+            final _Icon iOSIcon = newIcons.firstWhere((_Icon icon) => icon.id == '${ids[1]}$style');
+
+            platformAdaptiveDeclarations.add(_Icon.platformAdaptiveDeclaration('$flutterId$style', agnosticIcon, iOSIcon));
+          } catch (e) {
+            if (iconStyle == IconStyle.regular) {
+              stderr.writeln('Error while generating platformAdaptiveDeclarations: One or both of $ids not found.');
```

Comment:
Does this error message comes from the firstWhere throwing a IterableElementError? Instead of the catch, the `orElse` callback can be used to construct a more specific error message. For example (simplified):
1) First id not found
2) Second id not found
3) Both first and second id not found

Reply:
Done